daemon/daemon.go

```diff
@@ -536,38 +543,80 @@ func (d *Daemon) removeMasqRule() {
 	}
 }
 
-func (d *Daemon) installMasqRule() error {
+func (d *Daemon) installIptablesRules() error {
 	for _, c := range ciliumChains {
 		if err := c.add(); err != nil {
 			return fmt.Errorf("cannot add custom chain %s: %s", c.name, err)
 		}
 	}
 
-	// Masquerade all traffic from the host into the cilium_host interface
-	// if the source is not the internal IP
+	// Clear the Kubernetes masquerading mark bit to skip source PAT
+	// performed by kube-proxy for all packets destined for Cilium. Cilium
+	// installs a dedicated rule which does the source PAT to the right
+	// source IP.
+	if err := runProg("iptables", []string{
+		"-A", ciliumForwardChain,
+		"-o", "cilium_host",
+		"-m", "comment", "--comment", "cilium: clear masq bit for pkts to cilium_host",
+		"-j", "MARK", "--set-xmark", "0x0000/0x4000"}, false); err != nil {
+		return err
+	}
+
+	// kube-proxy does not change the default policy of the FORWARD chain
+	// which means that while packets to services are properly DNAT'ed,
+	// they are later dropped in the FORWARD chain. The issue has been
+	// resolved in #52569 and will be fixed in k8s >= 1.8. The following is
```

Comment:
With PR #52569 merged, will this workaround affect k8s behavior?

Reply:
The rules would not be needed in that case and they would be no-ops.